run_pacta_data_preparation.R

```diff
@@ -31,124 +31,127 @@ config <-
-factset_entity_financing_data_filename <- config$factset_entity_financing_data_filename
-factset_fund_data_filename <- config$factset_fund_data_filename
-factset_isin_to_fund_table_filename <- config$factset_isin_to_fund_table_filename
-factset_iss_emissions_data_filename <- config$factset_iss_emissions_data_filename
-factset_issue_code_bridge_filename <- config$factset_issue_code_bridge_filename
-factset_industry_map_bridge_filename <- config$factset_industry_map_bridge_filename
-factset_manual_pacta_sector_override_filename <- config$factset_manual_pacta_sector_override_filename
-update_currencies <- config$update_currencies
-export_sqlite_files <- config$export_sqlite_files
-export_archives <- config$export_archives
-imf_quarter_timestamp <- config$imf_quarter_timestamp
-pacta_financial_timestamp <- config$pacta_financial_timestamp
-market_share_target_reference_year <- config$market_share_target_reference_year
-time_horizon <- config$time_horizon
-scenario_sources_list <- config$scenario_sources_list
-sector_list <- config$sector_list
-other_sector_list <- config$other_sector_list
-zero_emission_factor_techs <- config$zero_emission_factor_techs
-green_techs <- config$green_techs
-scenario_raw_data_to_include <- config$scenario_raw_data_to_include
-tech_exclude <- config$tech_exclude
-scenario_geographies_list <- config$scenario_geographies_list
-global_aggregate_scenario_sources_list <- config$global_aggregate_scenario_sources_list
-global_aggregate_sector_list <- config$global_aggregate_sector_list
+expected_keys <- c(
+  "asset_impact_data_path",
+  "factset_data_path",
+  "data_prep_outputs_path",
+  "preflight_data_path",
+  "masterdata_ownership_filename",
+  "masterdata_debt_filename",
+  "ar_company_id__factset_entity_id_filename",
+  "factset_financial_data_filename",
+  "factset_entity_info_filename",
+  "factset_entity_financing_data_filename",
+  "factset_fund_data_filename",
+  "factset_isin_to_fund_table_filename",
+  "factset_iss_emissions_data_filename",
+  "factset_issue_code_bridge_filename",
+  "factset_industry_map_bridge_filename",
+  "factset_manual_pacta_sector_override_filename",
+  "update_currencies",
+  "export_sqlite_files",
+  "export_archives",
+  "imf_quarter_timestamp",
+  "pacta_financial_timestamp",
+  "market_share_target_reference_year",
+  "time_horizon",
+  "scenario_sources_list",
+  "sector_list",
+  "other_sector_list",
+  "zero_emission_factor_techs",
+  "green_techs",
+  "scenario_raw_data_to_include",
+  "tech_exclude",
+  "scenario_geographies_list",
+  "global_aggregate_scenario_sources_list",
+  "global_aggregate_sector_list"
+)
+stopifnot(all(expected_keys %in% names(config)))
 
 # create timestamped output directory
 system_timestamp <- format(
   Sys.time(),
   format = "%Y%m%dT%H%M%SZ",
   tz = "UTC"
 )
-data_prep_outputs_path <- file.path(
-  data_prep_outputs_path,
-  paste(pacta_financial_timestamp, system_timestamp, sep = "_")
+config[["data_prep_outputs_path"]] <- file.path(
```

Comment:
Since we're altering the values in `config` here, I think it would be prudent to capture the raw config, and also add that to `manifest.json`

```r
raw_config <- config::get(...)
config <- raw_config

#...
parameters <-
  list(
    config_name = config_name,
    config = unclass(config),
    raw_config = unclass(raw_config),
    input_filepaths = as.list(input_filepaths),

```

Reply:
done